From: Mohamad Al-Jaf <mohamadaljaf@gmail.com>
Subject: [PATCH v4 3/4] wdscore: Add stubs for ConstructPartialMsgV[AW].
Message-Id: <20220127232102.181453-3-mohamadaljaf@gmail.com>
Date: Thu, 27 Jan 2022 18:21:01 -0500
In-Reply-To: <20220127232102.181453-1-mohamadaljaf@gmail.com>
References: <20220127232102.181453-1-mohamadaljaf@gmail.com>

Enables the Windows MediaCreationTool21H2 to run.

Signed-off-by: Mohamad Al-Jaf <mohamadaljaf@gmail.com>
---
The debug channel emits a rather long string for msg:
fixme:wdscore:ConstructPartialMsgVW 67108864
L"\5343\7465\7075\614c\676e\6e49\4c69\616f\6564\3a72\
493a\696e\6974\6c61\7a69\3a65\4c20\4e41\2e47\4e49\2049\
6944\6572\7463\726f\2079\203d\7325" - stub

Should I get rid of it?
---
 dlls/wdscore/main.c             | 22 ++++++++++++++++++++++
 dlls/wdscore/wdscore.spec       |  4 ++--
 dlls/wdscore/wdscore_internal.h | 33 +++++++++++++++++++++++++++++++++
 3 files changed, 57 insertions(+), 2 deletions(-)
 create mode 100644 dlls/wdscore/wdscore_internal.h

diff --git a/dlls/wdscore/main.c b/dlls/wdscore/main.c
index 9001838431d..6f054cdd63b 100644
--- a/dlls/wdscore/main.c
+++ b/dlls/wdscore/main.c
@@ -24,6 +24,8 @@
 #include "wine/asm.h"
 #include "wine/debug.h"
 
+#include "wdscore_internal.h"
+
 WINE_DEFAULT_DEBUG_CHANNEL(wdscore);
 
 /***********************************************************************
@@ -52,3 +54,23 @@ void *CurrentIP(void)
     return 0;
 }
 #endif
+
+
+/***********************************************************************
+ *           ConstructPartialMsgVA (wdscore.@)
+ */
+LPVOID WINAPI ConstructPartialMsgVA( WdsLogLevel level, LPCSTR msg, va_list args )
+{
+    FIXME( "%u %s - stub\n", level, debugstr_a(msg) );
+    return NULL;
+}
+
+
+/***********************************************************************
+ *           ConstructPartialMsgVW (wdscore.@)
+ */
+LPVOID WINAPI ConstructPartialMsgVW( WdsLogLevel level, LPCWSTR msg, va_list args )
+{
+    FIXME( "%u %s - stub\n", level, debugstr_w(msg) );
+    return NULL;
+}
diff --git a/dlls/wdscore/wdscore.spec b/dlls/wdscore/wdscore.spec
index 282b6301688..b14778001a8 100644
--- a/dlls/wdscore/wdscore.spec
+++ b/dlls/wdscore/wdscore.spec
@@ -69,8 +69,8 @@
 #@ extern g_bEnableDiagnosticMode
 @ stub ConstructPartialMsgIfA
 @ stub ConstructPartialMsgIfW
-@ stub ConstructPartialMsgVA
-@ stub ConstructPartialMsgVW
+@ stdcall ConstructPartialMsgVA(long str ptr)
+@ stdcall ConstructPartialMsgVW(long wstr ptr)
 @ stdcall CurrentIP()
 @ stub EndMajorTask
 @ stub EndMinorTask
diff --git a/dlls/wdscore/wdscore_internal.h b/dlls/wdscore/wdscore_internal.h
new file mode 100644
index 00000000000..b5cbe3c8c25
--- /dev/null
+++ b/dlls/wdscore/wdscore_internal.h
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2022 Mohamad Al-Jaf
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
+ */
+
+#ifndef __WDSCORE_INTERNAL_H
+#define __WDSCORE_INTERNAL_H
+
+typedef enum _WdsLogLevel {
+    WdsLogLevelAssert     = 0x0000000,
+    WdsLogLevelFatalError = 0x1000000,
+    WdsLogLevelError      = 0x2000000,
+    WdsLogLevelWarning    = 0x3000000,
+    WdsLogLevelInfo       = 0x4000000,
+    WdsLogLevelStatus     = 0x5000000,
+    WdsLogLevelVerbose    = 0x6800000,
+    WdsLogLevelTrace      = 0x7000000
+} WdsLogLevel;
+
+#endif /* __WDSCORE_INTERNAL_H */

-- 
2.35.0

